### ISSUE TITLE
Question: Is it possible to open two windows with the same content and view?

### ISSUE THREAD

**Issue by dpshelio** [atlas-engineer/nyxt#451]
I wonder whether next can be used to show two windows with the same content and that they update at the same time.

I would like to be able to do so for when I need to do a presentation (for example when teaching python using jupyter notebooks) and there's not a secondary screen showing what's on the projector.

In the past, I used to do this using vnc of a single window. Now, with Wayland, I've not managed to find a solution for this. I thought Next could work, but on my simple attempt of opening another frame (window) - C-x 5-2 - and switching to the active buffer on the previous windows - C-x b -, it makes it disappear from the previous one. (I'm using v1.3.4 of next on a Gnome v3.34 in Archlinux).

Is this possible at all?

**Comment by jmercouris** [comment]
Hi thank you for your interest in Next!

It is not possible to view a buffer within two or more frames. The problem is unfortunately due to the underlying GTK/QT technologies. We could mirror the contents of one buffer into another buffer to achieve the same effect, but they would be in fact distinct buffers.

**Comment by Ambrevar** [comment]
This is a very exciting request, sadly it's a limitation of both the
WebKit and WebEngine renderers: they cannot render the same webview in
multiple viewports at the same time.  If you try to do it, only one
viewport will show the content while the others turn blank :(

I'd be very interested to know if there is a workaround! :)

**Comment by jmercouris** [comment]
@Ambrevar 

there *is* a workaround to enable rendering in two places, it basically involves copying the OpenGL Context of two widgets and showing it in two places, it is however very unstable, and only allows interaction with one of the views, the other is only a visual copy.

**Comment by Ambrevar** [comment]
Yup, I thought about something similar, but that wouldn't be very useful
because the common case for displaying the same buffer twice is that you
want to compare two different areas of the same buffer.

**Comment by dpshelio** [comment]
Thanks @jmercouris and @Ambrevar for the answers!
I am happy if one is only on 'view-mode' only, as that could be the one on the projector screen. Does copying the OpenGL context is something "simple" that a user can do or code in the config, or it needs to change the inners of Next? I know this is very niche, so not much hope that will become possible. I probably will get around it using VNC on an X11 desktop.
Thanks again! (ps. I let you decide whether to close this issue)

**Comment by Ambrevar** [comment]
For the projector use case, why not cloning the screen?

**Comment by dpshelio** [comment]
> For the projector use case, why not cloning the screen? 

Because I may have other content on my screen. For example a set of notes (like a speaker view).

When I teach I like to show the screen to the students free of any distraction, whereas on mine I want to have my notes, a timer, and other stuff that I may need to show during the class.
I wrote a [blog post](https://dpshelio.github.io/blog/2016/03/05/firefoxshare.html) a while ago that may illustrate better what I want to achieve. But since I moved to Wayland I've not been able to find a tool to replicate that with the browser. I was attracted to Next for two reasons for the cleaner, distraction-free, window and because I'm an emacs user. Then I thought that I could use this to clone a frame. 

My simplest solution would be to switch to a x11 desktop environment for teaching and use VNC as I used to, and that's probably what I'll do when I need to.
